python-backend/app/api/mines/mine/resources/mine.py

```diff
@@ -387,11 +393,26 @@ def put(self, mine_no_or_guid):
             mine.mine_location = MineLocation(
                 latitude=data['latitude'], longitude=data['longitude'])
             mine.save()
-            cache.delete(MINE_MAP_CACHE)
-        # Status validation
+
         _mine_status_processor(data.get('mine_status'), data.get('status_date'), mine)
+
+        cache.delete(MINE_MAP_CACHE)
+        self.call_rebuild_map_cache()
+
         return mine
 
+    def call_rebuild_map_cache(self):
+        app = current_app._get_current_object()
+        environ = request.environ
+
+        def do_work():
+            with app.request_context(environ):
+                mineMapResource = MineMapResource()
```

Comment:
What does the environ do here?

If you define rebuild_map_cache as a static method you may be able to skip the class init step.

Maybe more descriptive function names as well 😅 Alternatively you can just not call do_work and directly reference the rebuild_map_cache function in the thread initialization

Reply:
The environ is needed to pass in the request context.  My understanding is that this is how you give the thread access to the authentication information... as for more descriptive names how does 'do work faster' sound.